stdlib/public/Cxx/CxxSpan.swift

```diff
@@ -79,6 +79,7 @@ extension CxxSpan {
 
 @available(SwiftStdlib 6.1, *)
 extension Span {
+  @available(swift 6.2)
```

Comment:
This checks the language version (`-swift-version=xyz` flag), as opposed to compiler version, and a stdlib version.

I think we should check for one of the latter: the stdlib version if this depends on some feature of `Span`, or the compiler version if we're relying on new compiler behavior.

Reply:
I see. Unfortunately, we cannot check for the sodlib version because the scenario that breaks is using a newer stdlib with an older compiler. Do you know how to check for the compiler version (as opposed to the language version)?

Reply:
Ended up adding a language feature.